elastic/kibana
# [Vega] Allow faceted Vega-Lite charts to take correct size

Fixes https://github.com/elastic/kibana/issues/95291

It turns out that in the Vega-Lite internals, there are [4 types of specs](https://github.com/vega/vega-lite/blob/master/src/compile/buildmodel.ts), where `unit` and `layer` specs were previously working correctly but there were bugs in the Kibana handling of `facet` and `concat` specs.

<img width="1710" alt="Screen Shot 2021-06-24 at 2 54 05 PM" src="https://user-images.githubusercontent.com/666475/123324113-7554a880-d504-11eb-903f-0eb4c2b52803.png">

## Test instructions

Run kibana using the `vis_type_vega.enableExternalUrls: true` option in kibana.yml or command line flag. This lets you copy+past public Vega examples that use JSON datasets. For example, if you copy [this example](https://vega.github.io/vega-lite/examples/nested_concat_align.html) you will prepend `https://vega.github.io/vega-lite/examples/` in front of `data/movies.json` to get the full path for local use.

Behavior to test:

1. Vega-Lite [multi-view displays](https://vega.github.io/vega-lite/examples/#multi-view-displays) appear without warnings in the Kibana editor & dashboards. If you assign `width: 300`, it will affect the width of *each facet or repeated spec* without warnings. Please test both faceting & repeating specs, and test with pixel widths & with `autosize: none`
2. Vega-Lite [layered views](https://vega.github.io/vega-lite/examples/layer_histogram_global_mean.html) should render at the full size without scrollbars.
3. Vega should not be affected by this, but you can test a [Vega example](https://vega.github.io/vega/examples/) to make sure that it takes the full width by default.

### Checklist

Delete any items that are not applicable to this PR.
